mesh/pmesh.cpp

```diff
@@ -5568,6 +5568,97 @@ void ParMesh::PrintSharedEntities(const char *fname_prefix) const
    }
 }
 
+void ParMesh::GetGlobalVertexIndices(Array<HYPRE_Int> &gi)
+{
+   H1_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   gi.SetSize(GetNV());
+
+   Array<int> dofs;
+   for (int i=0; i<GetNV(); ++i)
+   {
+      fespace.GetVertexDofs(i, dofs);
+      gi[i] = fespace.GetGlobalTDofNumber(dofs[0]);
+   }
+}
+
+void ParMesh::GetGlobalEdgeIndices(Array<HYPRE_Int> &gi)
+{
+   if (Dim == 1)
+   {
+      GetGlobalVertexIndices(gi);
+      return;
+   }
+
+   ND_FECollection fec(1, Dim); // Order 1, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNEdges(),
+               "There should be a bijection between DOFs and edges");
+
+   gi.SetSize(GetNEdges());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNEdges(); ++i)
+   {
+      fespace.GetEdgeDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      const int ldof = (dofs[0] >= 0) ? dofs[0] : -1 - dofs[0];
+      gi[i] = fespace.GetGlobalTDofNumber(ldof);
+   }
+
+   MFEM_VERIFY(oneDof, "Each edge should have exactly one DOF");
+}
+
+void ParMesh::GetGlobalFaceIndices(Array<HYPRE_Int> &gi)
+{
+   if (Dim == 2)
+   {
+      GetGlobalEdgeIndices(gi);
+      return;
+   }
+   else if (Dim == 1)
+   {
+      GetGlobalVertexIndices(gi);
+      return;
+   }
+
+   RT_FECollection fec(0, Dim); // Order 0, mesh dimension (not spatial dimension).
+   ParFiniteElementSpace fespace(this, &fec);
+
+   MFEM_VERIFY(fespace.GetVSize() == GetNFaces(),
+               "There should be a bijection between DOFs and faces");
+
+   gi.SetSize(GetNFaces());
+
+   Array<int> dofs;
+   bool oneDof = true;
+   for (int i=0; i<GetNFaces(); ++i)
+   {
+      fespace.GetFaceDofs(i, dofs);
+      oneDof = oneDof && (dofs.Size() == 1);
+      const int ldof = (dofs[0] >= 0) ? dofs[0] : -1 - dofs[0];
+      gi[i] = fespace.GetGlobalTDofNumber(ldof);
+   }
+
+   MFEM_VERIFY(oneDof, "Each face should have exactly one DOF");
```

Comment:
Another check. If you insist keeping it, it might be better to change it to MFEM_ASSERT and move it inside the loop, to avoid the extra variable.